### PR TITLE
Fix install facilities and update Nix exp

### PR DESCRIPTION
**Pull request by GTrunSec**
- DONE for Emacs-Ng via Nix-build
image

https://github.com/GTrunSec/emacsNg-overlay/blob/main/flake.nix

### CODE REVIEW

**Comment by GTrunSec** [comment]
Nix flake default can be run commands as following:

- nix run github:emacs-ng/emacs-ng 
- nix build github:emacs-ng/emacs-ng

**Comment by black7375** [comment]
Maybe it will help Mac users as well too.
LGTM

**Comment by GTrunSec** [comment]
> Maybe it will help Mac users as well too.
sure, I'm going to test the nix-build on Darwin

**Comment by GTrunSec** [comment]
darwin build issue output(failure)

<details>
 Finished dev [unoptimized + debuginfo] target(s) in 4m 15s
clang -o temacs.tmp \
  -Demacs  -I. -I. -I../lib -I../lib      -I/nix/store/k9li2xib0l7mnwq8931vr2b2zf2z8gn7-librsvg-2.50.1-dev/include/librsvg-2.0 -I/nix/store/s1kk18fn59qn7s5sg22wgs84yrbc9hi2-glib-2.66.8-dev/include -I/nix/store/s1kk18fn59qn7s5sg22wgs84yrbc9hi2-glib-2.66.8-dev/include/glib-2.0 -I/nix/store/dib2bxf31w221iq8i0va40v5s87crpdw-glib-2.66.8/lib/glib-2.0/include -I/nix/store/zhc7sgxbnnvrv70mxxkp77nwa048mnaj-gdk-pixbuf-2.42.2-dev/include/gdk-pixbuf-2.0 -I/nix/store/2x5clzbkw6j5a3ybrvdlpki460pc19sf-cairo-1.16.0-dev/include/cairo -I/nix/store/vkc1c335nqd5brmf99qskfvpibf8hpcw-freetype-2.10.4-dev/include/freetype2 -I/nix/store/vkc1c335nqd5brmf99qskfvpibf8hpcw-freetype-2.10.4-dev/include -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/nix/store/fjb3x3ma6pqirpaxgz9zymadny71rxam-imagemagick-7.0.11-3-dev/include/ImageMagick-7 -I/nix/store/h7f7nja3yqmp5xqbnly9bm3lg5pg8vyd-zlib-1.2.11-dev/include -I/nix/store/pfs6100f64i48hmkmbchvgazis38f0x2-libpng-apng-1.6.37-dev/include/libpng16 -I/nix/store/f8zva2jl3pyx3gyswjbi9mjyaa021bza-libxml2-2.9.10-dev/include/libxml2        -I/nix/store/qshixhwg1rk4k6f92b4r1kc9fbxrz7a3-lcms2-2.12-dev/include       -MMD -MF deps/.d -MP  -I/nix/store/wvy7alwmbyifs0qbdq6li9lq7ggdbg0k-jansson-2.13.1/include -I/nix/store/85kl22j8hl5da2zc05vb4009yn829va9-gnutls-3.7.1-dev/include    -pthread -DEMACS_EXTERN_INLINE -Wno-switch -Wno-pointer-sign -Wno-string-plus-int -Wno-unknown-attributes -Wno-initializer-overrides -Wno-tautological-compare -Wno-tautological-constant-out-of-range-compare -DMAC_OS_X_VERSION_MAX_ALLOWED=101200 -O  -framework AppKit -framework IOKit -framework Carbon -framework IOSurface -L.././rust_src/target/debug -framework Security \
    dispnew.o frame.o scroll.o xdisp.o menu.o  window.o charset.o coding.o category.o ccl.o character.o chartab.o bidi.o cm.o term.o terminal.o xfaces.o    emacs.o keyboard.o macros.o keymap.o sysdep.o bignum.o buffer.o filelock.o insdel.o marker.o minibuf.o fileio.o dired.o cmds.o casetab.o casefiddle.o indent.o search.o regex-emacs.o undo.o alloc.o pdumper.o data.o doc.o editfns.o callint.o eval.o floatfns.o fns.o font.o print.o lread.o emacs-module.o syntax.o  bytecode.o comp.o dynlib.o process.o gnutls.o callproc.o region-cache.o sound.o timefns.o atimer.o doprnt.o intervals.o textprop.o composite.o xml.o lcms.o kqueue.o  profiler.o decompress.o thread.o systhread.o        fontset.o fringe.o image.o  json.o nsterm.o nsfns.o nsmenu.o nsselect.o nsimage.o macfont.o -lemacsng terminfo.o lastfile.o      ../lib/libgnu.a       -ltiff -ljpeg -L/nix/store/rpgg1flx97hsks915l8ikqn8yhm0nihv-zlib-1.2.11/lib -L/nix/store/gw5d1ry3jphqs7mpz9klcwsdihjnfa0b-libpng-apng-1.6.37/lib -lpng16 -lz     -L/nix/store/8dv0ynqaak5ccb5gy96irvgsmgdpxh35-librsvg-2.50.1/lib -L/nix/store/dib2bxf31w221iq8i0va40v5s87crpdw-glib-2.66.8/lib -L/nix/store/z80xm236r8p88qa2pkfbf6y13cg5876v-gdk-pixbuf-2.42.2/lib -L/nix/store/2j5f20f5q5dqihcksi4bsj36pdvs0lxk-cairo-1.16.0/lib -lrsvg-2 -lm -lgio-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -lcairo -L/nix/store/acl4201lhs4wb264zr0h8gdhhsjpbrjb-imagemagick-7.0.11-3/lib -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI            -L/nix/store/fgxj11w6jkmf1z1lvi29lvxsm978fc28-libxml2-2.9.10/lib -lxml2     -lncurses         -L/nix/store/a3dnvzpvghm1lkqlww6wb9hj1spb5whk-gnutls-3.7.1/lib -lgnutls   -L/nix/store/89l52r42c21p02jn9xry2acvcnx3r6b9-lcms2-2.12/lib -llcms2   -lz   -L/nix/store/wvy7alwmbyifs0qbdq6li9lq7ggdbg0k-jansson-2.13.1/lib -ljansson -lgmp -lgccjit -ldl -ldl -lm -lresolv
Undefined symbols for architecture x86_64:
  "_AREF", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_ASET", referenced from:
      emacsng::parsing::serde_to_lisp::hf6864aa02a93e6ab in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_ASIZE", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_BYTE_POS_ADDR", referenced from:
      emacsng::ng_module::ng_module_access_current_buffer_contents::hf9211221b8313019 in libemacsng.a(emacsng-d31899dfbc0ab4ca.34rmcw0kldcx98jx.rcgu.o)
  "_CHECK_IMPURE", referenced from:
      emacs::list::LispCons::check_impure::hb5b3df6979516514 in libemacsng.a(emacs-9ba44f5acbd6264b.4o7gk136jyuaz7g4.rcgu.o)
  "_CHECK_NUMBER", referenced from:
      emacsng::javascript::js_set_tick_rate::hf50ce9623d646c5f in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
  "_FIXNUMP", referenced from:
      emacsng::javascript::lisp_invoke::h2b5e0a1ee47b6146 in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::javascript::js_reenter_inner::_$u7b$$u7b$closure$u7d$$u7d$::h95e2207e86ebd8ec in libemacsng.a(emacsng-d31899dfbc0ab4ca.3cmonssftfcmpu4z.rcgu.o)
  "_FLOATP", referenced from:
      emacsng::javascript::lisp_invoke::h2b5e0a1ee47b6146 in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::javascript::permissions_from_args::hc1fc995adeb2daab in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::javascript::js_reenter_inner::_$u7b$$u7b$closure$u7d$$u7d$::h95e2207e86ebd8ec in libemacsng.a(emacsng-d31899dfbc0ab4ca.3cmonssftfcmpu4z.rcgu.o)
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_HASH_KEY", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_HASH_TABLE_P", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_HASH_TABLE_SIZE", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_HASH_VALUE", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_INTEGERP", referenced from:
      emacsng::javascript::js_reenter_inner::h673d48e493824d5f in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_NILP", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_STRINGP", referenced from:
      emacsng::javascript::lisp_invoke::h2b5e0a1ee47b6146 in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::javascript::js_reenter_inner::_$u7b$$u7b$closure$u7d$$u7d$::h95e2207e86ebd8ec in libemacsng.a(emacsng-d31899dfbc0ab4ca.3cmonssftfcmpu4z.rcgu.o)
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_SYMBOLP", referenced from:
      emacsng::parsing::lisp_to_serde::_$u7b$$u7b$closure$u7d$$u7d$::h3731c3b70519c9d1 in libemacsng.a(emacsng-d31899dfbc0ab4ca.5c93z2r6sajyrmv8.rcgu.o)
  "_SYMBOL_NAME", referenced from:
      emacsng::parsing::lisp_to_serde::_$u7b$$u7b$closure$u7d$$u7d$::h3731c3b70519c9d1 in libemacsng.a(emacsng-d31899dfbc0ab4ca.5c93z2r6sajyrmv8.rcgu.o)
  "_USER_PTRP", referenced from:
      emacs::lisp::LispObject::is_user_ptr::he5441ed8219bb6bf in libemacsng.a(emacs-9ba44f5acbd6264b.3r7bmv6s6bfjseaj.rcgu.o)
  "_VECTORP", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_XFLOATINT", referenced from:
      emacsng::javascript::js_set_tick_rate::hf50ce9623d646c5f in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
  "_XFLOAT_DATA", referenced from:
      emacsng::javascript::permissions_from_args::hc1fc995adeb2daab in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_XHASH_TABLE", referenced from:
      emacsng::parsing::lisp_to_serde::h316cd13996baeecc in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
      emacsng::parsing::serde_to_lisp::hf6864aa02a93e6ab in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_XUSER_PTR", referenced from:
      emacs::lisp::LispObject::as_userdata_ref::h4ac9f669adefba7b in libemacsng.a(emacsng-d31899dfbc0ab4ca.1qb20j30756qxv4d.rcgu.o)
      emacs::lisp::LispObject::as_userdata_ref::hcb0d8d8649c5660a in libemacsng.a(emacsng-d31899dfbc0ab4ca.1qb20j30756qxv4d.rcgu.o)
      emacs::lisp::LispObject::as_userdata_ref::hd0e928a0572918e8 in libemacsng.a(emacsng-d31899dfbc0ab4ca.1qb20j30756qxv4d.rcgu.o)
      emacsng::ng_async::to_owned_userdata::hb69fa3beb134e82d in libemacsng.a(emacsng-d31899dfbc0ab4ca.2oecz8svsxva8zdi.rcgu.o)
      emacsng::ng_async::internal_send_message::hb527bdc1c164f8fc in libemacsng.a(emacsng-d31899dfbc0ab4ca.2oecz8svsxva8zdi.rcgu.o)
  "_build_string", referenced from:
      _$LT$emacs..lisp..LispObject$u20$as$u20$core..convert..From$LT$$RF$str$GT$$GT$::from::hb841304f9c7c6746 in libemacsng.a(emacs-9ba44f5acbd6264b.3r7bmv6s6bfjseaj.rcgu.o)
      emacsng::ng_async::EmacsPipe::create::hf6dbb48462a1efde in libemacsng.a(emacsng-d31899dfbc0ab4ca.2oecz8svsxva8zdi.rcgu.o)
  "_frame_dimension", referenced from:
      emacs::frame::_$LT$impl$u20$emacs..lisp..ExternalPtr$LT$emacs..bindings..frame$GT$$GT$::internal_border_width::hb8f2ab6b8f21a814 in libemacsng.a(emacs-9ba44f5acbd6264b.3r7bmv6s6bfjseaj.rcgu.o)
  "_intern_c_string", referenced from:
      emacsng::ng_async::EmacsPipe::create::hf6dbb48462a1efde in libemacsng.a(emacsng-d31899dfbc0ab4ca.2oecz8svsxva8zdi.rcgu.o)
     (maybe you meant: _intern_c_string_1)
  "_make_fixed_natnum", referenced from:
      emacsng::parsing::serde_to_lisp::hf6864aa02a93e6ab in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
  "_make_fixnum", referenced from:
      emacsng::javascript::lisp_make_finalizer::h42b5ac9b7cdd6100 in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::javascript::lisp_make_lambda::h31b975a9d7fe1a4f in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
      emacsng::javascript::lisp_fixnum::h9d4c594fa448d538 in libemacsng.a(emacsng-d31899dfbc0ab4ca.2onyh3l4rtp670i1.rcgu.o)
  "_make_int", referenced from:
      emacsng::parsing::serde_to_lisp::hf6864aa02a93e6ab in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
     (maybe you meant: _make_integer_mpz, _make_interval )
  "_make_lisp_symbol", referenced from:
      emacs::symbol::_$LT$impl$u20$core..convert..From$LT$emacs..lisp..ExternalPtr$LT$emacs..bindings..Lisp_Symbol$GT$$GT$$u20$for$u20$emacs..lisp..LispObject$GT$::from::hb329f2828342dd8a in libemacsng.a(emacs-9ba44f5acbd6264b.3r7bmv6s6bfjseaj.rcgu.o)
  "_make_uint", referenced from:
      emacsng::parsing::serde_to_lisp::hf6864aa02a93e6ab in libemacsng.a(emacsng-d31899dfbc0ab4ca.3xwrexuiqb7khzoy.rcgu.o)
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:743: temacs] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-emacsNg.drv-0/source/src'
make: *** [Makefile:432: src] Error 2
error: --- Error ---------------------------------------------------------------------------------------------- nix-build
builder for '/nix/store/vx8qrddhp1dxmcfwbj778j7kd7si5qrr-emacsNg.drv' failed with exit code 2
┌─[gtrun@macOSdeMBP] - [~/src/emacsNg-overlay] - [Fri Apr 09, 02:40]
└─[$] <git:(main)> 
┌─[gtrun@macOSdeMBP] - [~/src/emacsNg-overlay] - [Fri Apr 09, 02:40]
└─[$] <git:(main)> 
<details>

File: build_emacs_ng.sh [hunk 1]
```diff
@@ -1,8 +1,12 @@
 sudo add-apt-repository -y ppa:ubuntu-toolchain-r/ppa
+sudo apt update
 sudo apt install -y autoconf make checkinstall texinfo libxpm-dev libjpeg-dev \
      libgtk-3-dev libgif-dev libtiff-dev libpng-dev libgnutls28-dev libncurses5-dev \
-     libsystemd-dev libjansson-dev libharfbuzz-dev libgccjit-10-dev gcc-10 g++-10
-sudo apt update
+     libsystemd-dev libjansson-dev libharfbuzz-dev libgccjit-10-dev gcc-10 g++-10 libxt-dev \
+     libclang-10-dev 
+
+sudo snap install rustup --classic
```
Comment (kiennq) [inline build_emacs_ng.sh:8]: The release workflow has already install rust via Gihub action.
https://github.com/emacs-ng/emacs-ng/blob/415de78e6d4fdd2c7f3f28cafe37ac1be710ee41/.github/workflows/release.yml#L20

Can you remove that step there?

Also, `snap` is fairly new installation method and not available on all distro. Can you use something like instead?
```suggestion
 curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --profile minimal
```

Comment (GTrunSec) [reply build_emacs_ng.sh:8]: Thanks, I only tested snap on ubuntu.

**Review by ericdallo — APPROVED** [review]
LGTM, thank you!

Do you think in move the overlay code to the [emacs common overlay](https://github.com/nix-community/emacs-overlay) ? This way nix users would follow the same pattern and use the same overlay but changing to emacsNg

**Comment by ericdallo** [comment]
BTW @GTrunSec , I'm getting a out of space on device when compiling with `nix-build` but I have 194gb available on my device:
image

**Comment by GTrunSec** [comment]
@ericdallo 
check the solution https://github.com/NixOS/nixops/issues/1402#issuecomment-770181921

File: docs/getting-started.md [hunk 2]
```diff
@@ -53,6 +53,65 @@ compile it(no guarantee that they work):
...
+
+- using `cachix` to (download binary cache)speed up build
+
+```bash
+nix-env -iA cachix -f https://cachix.org/api/v1/install #install cachix
+cachix use emacsng
+#then
+nix-build
+#or
+nix build github:emacs-ng/emacs-ng
```
Comment (ericdallo) [inline docs/getting-started.md:76]:
```suggestion
nix build -o github:emacs-ng/emacs-ng
```

**Review by ericdallo — APPROVED** [review]
Tested it manually and it's working great!
Thank you very much for this great contribution!

**Pull request merged**